mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/task/SolveMailboxInconsistenciesService.java

```diff
@@ -329,10 +407,25 @@ private void assertValidVersion() {
     private Flux<Result> processMailboxDaoInconsistencies(Context context) {
         return mailboxDAO.retrieveAllMailboxes()
             .flatMap(this::detectInconsistency)
-            .flatMap(inconsistency -> inconsistency.fix(context, mailboxDAO, mailboxPathV2DAO))
+            .collect(Guavate.toImmutableList())
+            // Wait to ensure concurrentWrite upon diagnostic had time to complete
+            .delayElement(gracePeriod)
+            .flatMapMany(Flux::fromIterable)
+            .flatMap(inconsistency -> fixInconsistencyIfNeeded(context, inconsistency))
             .doOnNext(any -> context.processedMailboxEntries.incrementAndGet());
     }
 
+    private Mono<Result> fixInconsistencyIfNeeded(Context context, Inconsistency inconsistency) {
```

Comment:
Tests are missing regarding this right?

Reply:
Seems there -> #3130, right? Was not clear though :)

Reply:
Yes, I did not wanted to link the entire PR to the *Cassandra session instumentation topic*

But I was also unwilling to contribute broken code. Hence the choice of putting untested code here.